### PR TITLE
[SPARK-40890][SQL][TESTS] Check error classes in DataSourceV2SQLSuite

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This PR aims to replace 'intercept' with 'Check error classes' in DataSourceV2SQLSuite.

### Why are the changes needed?
The changes improve the error framework.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
By running the modified test suite:
```
$ build/sbt "test:testOnly *DataSourceV2SQLSuite"
```

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @MaxGekk

File: sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala [hunk 1]
```diff
@@ -2428,23 +2719,38 @@ class DataSourceV2SQLSuiteV1Filter extends DataSourceV2SQLSuite with AlterTableT
...
-    assert(e.message.contains(s"$sqlCommand is not supported for v2 tables"))
+    checkError(
+      exception = intercept[AnalysisException] {
+        sql(s"$sqlCommand $sqlParams")
+      },
+      errorClass = "_LEGACY_ERROR_TEMP_1124",
+      parameters = Map("cmd" -> sqlCommand))
   }
 
   private def assertAnalysisErrorClass(
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala:2730]: Let's remove `assertAnalysisErrorClass()`, and use `checkError()` directly. You could introduce an method `analysisException` as we did before for ParseException

File: sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala [hunk 2]
```diff
@@ -63,11 +65,15 @@ abstract class DataSourceV2SQLSuite
...
-    assert(ex.getMessage.contains(expectedError))
+      errorClass: String,
+      parameters: Map[String, String]): Unit = {
+    checkError(
+      exception = intercept[AnalysisException] {
+        sql(sqlStatement)
+      },
+      errorClass = errorClass,
+      parameters = parameters
+    )
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala:76]: Let's introduce a helper method for this:
```scala
      exception = intercept[AnalysisException] {
        sql(sqlStatement)
      },
```
and invoke `checkError` directly.

Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala]: Done

File: sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala (hunk 1 above)
Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala]: Done

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by MaxGekk** [comment]
@panbingkun Please, fix the code style issue:
```
[error] /home/runner/work/spark/spark/sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala: Expected token RPAREN but got Token(EQUALS,=,76130,=)
```

**Comment by panbingkun** [comment]
> @panbingkun Please, fix the code style issue:
> 
> ```
> [error] /home/runner/work/spark/spark/sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala: Expected token RPAREN but got Token(EQUALS,=,76130,=)
> ```

Done.

**Comment by MaxGekk** [comment]
It seems this is related to your changes:
```
DataSourceV2SQLSuiteV1Filter.MERGE INTO TABLE
org.scalatest.exceptions.TestFailedException: "_LEGACY_ERROR_TEMP_2309" did not equal null
```
Most likely after https://github.com/apache/spark/commit/3c967f06e6c37360e53f9c7a5ecff95ae818e713, need to merge/rebase the master.

**Comment by panbingkun** [comment]
> It seems this is related to your changes:
> 
> ```
> DataSourceV2SQLSuiteV1Filter.MERGE INTO TABLE
> org.scalatest.exceptions.TestFailedException: "_LEGACY_ERROR_TEMP_2309" did not equal null
> ```
> 
> Most likely after [3c967f0](https://github.com/apache/spark/commit/3c967f06e6c37360e53f9c7a5ecff95ae818e713), need to merge/rebase the master.

Done, waiting for CI.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master. I ran the test locally.
Thank you, @panbingkun.

**Pull request closed without merging**